Mathlib/Tactic/MinImports.lean

```diff
@@ -37,20 +37,65 @@ def getVisited (env : Environment) (decl : Name) : NameSet :=
 
 /-- `getId stx` takes as input a `Syntax` `stx`.
 If `stx` contains a `declId`, then it returns the `ident`-syntax for the `declId`.
+If `stx` is a nameless instance, then it also tries to fetch the `ident` for the instance.
 Otherwise it returns `stx` itself. -/
-def getId (stx : Syntax) : Syntax :=
+def getId (stx : Syntax) : CommandElabM Syntax := do
+  -- if the command contains a `declId`, we use the implied `ident`
   match stx.find? (·.isOfKind ``Lean.Parser.Command.declId) with
-    | some declId => declId[0]
-    | none => stx
-
-/--`getAllImports cmd` takes a `Syntax` input `cmd` and returns the `NameSet` of all the
-module names that are implied by the `SyntaxNodeKinds` and the identifiers contained in `cmd`. -/
-def getAllImports {m : Type → Type} [Monad m] [MonadResolveName m] [MonadEnv m]
-    (cmd : Syntax) (dbg? : Bool := false) :
-    m NameSet := do
+    | some declId => return declId[0]
+    | none =>
+      -- otherwise, there is the possibility that the command is a nameless `instance`
+      match stx.find? (·.isOfKind ``Lean.Parser.Command.instance) with
+                | none => return stx
+                | some stx => do
+                  -- if it is a nameless `instance`, we retrieve the autogenerated name
```

Comment:
Also optional: 
```suggestion
                  -- If it is a nameless `instance`, we retrieve the autogenerated name.
```

or
```suggestion

```suggestion
                  -- In that case, we retrieve the autogenerated name.